add TI am64xx sk support (v2)

mike-scott opened the pull request:
- new linux-lmp-ti-staging kernel recipe to support TI's kernel build using defconfig_builder.sh + kmeta data
- u-boot script for u-boot-ostree-scr
- uboot-ti-staging patch to raise CONFIG_SYS_BOOTM_LEN to 64MB
- LmP board settings for am64xx-sk
- bump kmeta-linux-lmp-5.4.y SHA to 71b6bbf223ba510a6c24c62418dfbcc939deb1d6

mike-scott commented:
@Tim-Anderson apologies for closing your previous PR in order to push this revised patch set.  Let me know what you think (specifically the kernel recipe).

ricardosalveti commented on meta-lmp-bsp/recipes-bsp/u-boot/u-boot-base-scr/am64xx-sk/boot.cmd:
```diff
@@ -0,0 +1,8 @@
+setenv devtype mmc
+setenv devnum ${mmcdev}
+run findfdt
+load ${devtype} ${devnum}:2 ${loadaddr} /boot/loader/uEnv.txt
```
Do we really have /boot/loader/uEnv.txt with lmp-base? In theory that file is ostree specific.

mike-scott replied:
Dropped this recipe.  It wasn't needed and it was broken.

ricardosalveti commented on meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr/am64xx-sk/boot.cmd:
```diff
@@ -0,0 +1,8 @@
+load ${devtype} ${devnum}:2 ${loadaddr} /boot/loader/uEnv.txt
+env import -t ${loadaddr} ${filesize}
+load ${devtype} ${devnum}:2 ${loadaddr} ${kernel_image}
+bootm ${loadaddr}#conf@ti_${fdtfile}
```
boot.cmd should really just have enough for loading uEnv.txt (from vfat).

e.g. with freedom:

```
$ cat ./meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr/freedom-u540/boot.cmd
if test ${distro_bootpart} != 1
then
    echo "Boot partition needs to be the first partition"
    exit
fi

fatload ${devtype} ${devnum}:1 $scriptaddr /uEnv.txt
env import -t $scriptaddr $filesize
run bootcmd

$ cat ./meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr/freedom-u540/uEnv.txt.in
bootcmd_otenv=load ${devtype} ${devnum}:3 ${scriptaddr} /boot/loader/uEnv.txt; env import -t ${scriptaddr} ${filesize}
bootcmd_load_f=load ${devtype} ${devnum}:3 ${ramdisk_addr_r} "/boot"${kernel_image}
bootcmd_run=bootm ${ramdisk_addr_r}
bootcmd=run bootcmd_otenv; run bootcmd_load_f; run bootcmd_run
```

mike-scott commented:
- Rebased
- Dropped u-boot-base-scr as u-boot BOOTCOMMAND is hard-coded
- Made sure the kernel recipe deploys the DTB to rootfs boot/ dir to that default BOOTCOMMAND works for lmp-base

mike-scott commented:
@ricardosalveti I think this leaves me to look at the u-boot-ostree-scr files and make sure they are in sync with the other machines.

Tim-Anderson commented:
I have moved these edits into PR #270 so I am closing this PR
